Ginger/Ginger/Drivers/DriversWindows/MobileDriverWindow.xaml.cs

```diff
@@ -1054,9 +1529,96 @@ private void PerformScreenSwipe(eSwipeSide swipeSide, double impact = 1)
         }
         #endregion Functions
 
-        private void xRecordingImage_Click(object sender, RoutedEventArgs e)
+        GenericWindow genWin;
+
+        private void OpenPopUpWindow(string content, string title)
         {
+            string tempFilePath = GingerCoreNET.GeneralLib.General.CreateTempTextFile(content);
+            //string filePath = Path.Combine(Directory.GetCurrentDirectory(), "temp.json");
+            //File.WriteAllText(filePath, content);
+            if (System.IO.File.Exists(tempFilePath))
+            {
+                Application.Current.Dispatcher.Invoke(() =>
+                {
+                    DocumentEditorPage docPage = new DocumentEditorPage(tempFilePath, enableEdit: false, UCTextEditorTitle: string.Empty);
+                    docPage.Width = 800;
+                    docPage.Height = 800;
+                    GingerCore.General.LoadGenericWindow(ref genWin, App.MainWindow, eWindowShowStyle.Free, title, docPage);
+
+                });
+
+                System.IO.File.Delete(tempFilePath);
+                return;
+            }
+        }
+
+
 
+        private void xAgentBtn_Click(object sender, RoutedEventArgs e)
+        {
+            Application.Current.Dispatcher.Invoke(() =>
+            {
+                AgentEditPage agentEditPage = new AgentEditPage(mAgent);
+                agentEditPage.Width = 800;
+                agentEditPage.Height = 800;
+                ((Grid)((ScrollViewer)agentEditPage.Content).Content).IsEnabled = false;
+                GingerCore.General.LoadGenericWindow(ref genWin, App.MainWindow, eWindowShowStyle.Free, mAgent.Name, agentEditPage);
+            });
+        }
+
+        private void xRamUsageBtn_Click(object sender, RoutedEventArgs e)
+        {
+            OpenPopUpWindow(DictionaryToString(mDeviceMemoryInfo).Replace(", ", Environment.NewLine), "Full device's Memory Information");
+        }
+
+        private void xNetworksBtn_Click(object sender, RoutedEventArgs e)
+        {
+            OpenPopUpWindow(mDeviceNetworkInfo, "Full device's Network Information");
+        }
+
+
+
+        //private void xAgentValueLbl_TargetUpdated(object sender, System.Windows.Data.DataTransferEventArgs e)
+        //{
+        //    string labelContent = xAgentValueLbl.Content.ToString();
+        //    if (labelContent.Length > 35)
+        //    {
+        //        xAgentValueLbl.Content = labelContent.Substring(0, 35) + "...";
+        //    }
+        //}
+
+        private void xExtraInfoBtn_ExtraInfo_Click(object sender, RoutedEventArgs e)
+        {
+            DeviceInfo deviceInfo = (DeviceInfo)((ucButton)sender).DataContext;
```

Comment:
i didn't get why this method is needed if you already having event handling for each extra info click above.
just add to DeviceInfo field which will store the event and trigger it on click

Reply:
I removed to above individual functions and kept this function instead.